R/apply_duration_scale.R

```diff
@@ -0,0 +1,47 @@
+#' Apply infusion duration scale to a regimen
+#'
+#' E.g. see Centanni et al. Clin Pharmacokinet 2024. An estimated scaling factor
+#' for the length of the infusion was applied there in a model for vincristine.
+#' This is likely most relevant for very short infusions.
+#'
+#' Implementation is similar to handling of `lagtime`, i.e. the regimen that is the
+#' input for the simulation function is updated.
+#'
+#' @param regimen PKPDsim regimen
+#' @param duration_scale infusion length scale.
+#' @param parameters parameter list, required if the duration scale is
+#' specified as a parameter.
+#'
+#' @export
+#'
+#' @return Original regimen with infusion lengths scaled by a factor
+#'
+apply_duration_scale <- function(
+  regimen,
+  duration_scale = NULL,
+  parameters = NULL
+) {
+  if(is.null(regimen$t_inf) || all(regimen$t_inf == 0)) {
+    warning("`duration_scale` is only relevant for infusion regimens with an infusion length > 0.")
+    return(regimen)
+  }
+  if(!is.null(duration_scale)) {
+    if(class(duration_scale) %in% c("numeric", "integer")) {
+      if(length(duration_scale) == 1) {
+        regimen$t_inf <- regimen$t_inf * duration_scale
+      } else {
+        regimen$t_inf <- regimen$t_inf * duration_scale[regimen$cmt]
```

Comment:
```suggestion
        regimen$t_inf <- regimen$t_inf * duration_scale[regimen$type]
```

does it make more sense to use the `type` argument since i think that is how we tend to differentiate doses of different types (which go to different compartments)

Reply:
Sorry, I misread my earlier code. That 5 lines of code I removed is actually useful, at least when we're using a vector of duration scales that matches `cmt`s, which we do. The removed code ensures there is always a `cmt` in the regimen, even when not specified by the user. So will bring that code back. I think it is OK where it is, and doesn't need to go on top of the function, since we only want to make a modification to the regimen if its actually needed, IMO.

Regarding your 2nd question whether we should link it to `type` rather than `cmt`: I think it's less consistent to link it to `type`: we now have "lagtime" also linked to a specific `cmt`, not to an observation type. I see those as two very similar features. Also in NONMEM durations and infusion lengths are linked to compartments, so for most users more intuitive. An option for a future ticket could be that `lagtime` and `duration_scale` could also be specified as a list, where they then indeed can be mapped to observation types.